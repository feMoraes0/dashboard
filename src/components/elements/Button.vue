<template>
  <button type='button' :class='style'>
    <slot />
  </button>
</template>

<script>
  export default {
    name: 'Button',
    props: {
      type: {
        type: String,
        default: 'button'
      }
    },
    computed: {
      style() {
        return this.type === 'button' ? 'btn-primary' : 'btn-link';
      }
    }
  }
</script>

<style scoped>
  button {
    position: relative;
    width: 100%;
    padding-top: 15px;
    padding-bottom: 15px;
    padding-left: 23px;
    border-radius: 8px;
    border: none;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    cursor: pointer;
    font-size: 15px;
    font-weight: 400;
    line-height: 16px;
    color: white;
  }

  button svg {
    width: 18px !important;
    height: 18px !important;
    color: white;
    margin-right: 27px;
  }

  .btn-primary {
    background-color: #0054FE;
  }

  .btn-link {
    background-color: transparent;
    color: #74798C !important;
    font-weight: 500 !important;
  }

  .btn-link svg {
    color: #74798C !important;
  }
</style>