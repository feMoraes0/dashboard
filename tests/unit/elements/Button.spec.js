import { shallowMount } from '@vue/test-utils';
import Button from '../../../src/components/elements/Button.vue';

let wrapper;

const factory = (propsData) => {
  return shallowMount(Button, {
    propsData: {
      ...propsData
    }
  });
}

describe('Button Component', () => {
  beforeEach(() => {
    wrapper = factory();
  });
  
  afterEach(() => {
    wrapper.destroy();
  });

  it('has property with default value', () => {
    // THEN
    expect(wrapper.vm.$props).toEqual({'type': 'button'});
  });

  it('should return btn-primary class', () => {
    // THEN
    expect(wrapper.vm.buttonClass).toEqual('btn-primary');
  });

  describe('with not default property', () => {
    beforeEach(() => {
      wrapper = factory({
        type: 'link'
      });
    });

    afterEach(() => {
      wrapper.destroy();
    });

    it('has property modified', () => {
      // THEN
      expect(wrapper.vm.$props).toEqual({'type': 'link'});
    });

    it('should return btn-link class', () => {
      // THEN
      expect(wrapper.vm.buttonClass).toEqual('btn-link');
    });
  })
});