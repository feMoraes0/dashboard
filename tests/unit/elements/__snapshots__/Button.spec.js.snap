// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render to a snapshot when component is initialized 1`] = `
<button
  class="btn-primary"
  type="button"
/>
`;
